cluster-scope/base/apiextensions.k8s.io/customresourcedefinitions/reports.curatoroperator.io/customresourcedefinition.yaml

```diff
@@ -0,0 +1,175 @@
+spec:
+  group: curator.operatefirst.io
+  names:
+    kind: Report
+    listKind: ReportList
+    plural: reports
+    singular: report
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: Report is the Schema for the reports API
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: ReportSpec defines the desired state of Report
+            properties:
+              cronjobNamespace:
+                description: Namespace in which cron job is created
+                type: string
+              databaseHostName:
+                description: Value for the Database HostName Environment Variable
+                type: string
+              databaseName:
+                description: Value for the Database Name Environment Variable
+                type: string
+              databasePassword:
+                description: Value for the Database Password Environment Variable
+                type: string
+              databasePort:
+                description: Value for the Database Environment Variable in order
+                  to define the port which it should use. It will be used in its container
+                  as well
+                type: string
+              databaseUser:
+                description: Value for the Database User Environment Variable
+                type: string
+              emailPassword:
+                description: Sender email password which does not need 2FA
+                type: string
+              emailRecepients:
+                description: Recepients email who will receive the reports, should
+                  be in the format {"user1@gmail.com":{"cc":["cc1@gmail.com"]}}
+                type: string
+              emailReports:
+                description: If email has to be sent then this value should be set
+                  to true
+                type: boolean
+              emailUser:
+                description: Sender email address which will be used as a mail address
+                type: string
+              reportFrequency:
+                description: Frequency of the report generation
```

Comment:
What are the expected values for this field? Is it every X units of time or is it a descriptive word like "Daily", "Weekly", etc?

Reply:
Yes, It's daily weekly and monthly. It's is specified in the sample CR - https://github.com/operate-first/curator-operator/blob/94dd0fe6eca5604f664458eac2f2a2acc631cace/config/samples/curator_v1alpha1_report.yaml#L13

If required, I will also make changes in it to accept only that three values.

Reply:
Not required to make it only accept those 3 values, but I would appreciate the description to be update to describe the expected values, since not everyone may have access to the examples, or know how to find them (I didn't.)

Reply:
Sure. I would update the description.

Reply:
Updated in the latest commit.